### ISSUE TITLE
Make `rsDataObjUnlink` warning less noisy

### ISSUE THREAD

**Issue by alanking** [irods/irods#7953]
- [x] main
- [x] 4-3-stable

---

## Feature

`rsDataObjUnlink` prints something like the following warning message when a replica is being targeted that is not in an iRODS resource vault:

> Replica is not in a vault. Unregistering replica and leaving it on disk as-is [data_object=/tempZone/home/rods/ufs/dir09/file08, physical_object=/data/ufs/dir09/file08, vault_path=/var/lib/irods/Vault].

Here is the code:
https://github.com/irods/irods/blob/bc2a9bf0d8d3587d1aad8bf6a9b3db650e0d6944/server/api/src/rsDataObjUnlink.cpp#L543-L551

The `oprType` is being set to `UNREG_OPR`. This happens even if the `oprType` is already `UNREG_OPR`.

This is observed when using the python-irodsclient's `DataObjectManager.unregister` method:
https://github.com/irods/python-irodsclient/blob/527417da41a997198e514e0e211b5422585a97a0/irods/manager/data_object_manager.py#L620-L628

As you can see, `OPR_TYPE_KW` is being set in the `**options` which is then used in `DataObjectManager.unlink` to set the `oprType` in the API call:
https://github.com/irods/python-irodsclient/blob/527417da41a997198e514e0e211b5422585a97a0/irods/manager/data_object_manager.py#L597-L611

Marking as a feature / enhancement because it's not really a bug - just annoying. Willing to upgrade to bug.

**Comment by trel** [comment]
There's no change in behavior that you're proposing... right?

So... setting it to `logger::api::debug` is perhaps sufficient to not be as annoying?

**Comment by alanking** [comment]
No, we want to show the warning in the event that we attempt to unlink a replica which is not in an iRODS resource vault because we are changing the behavior requested by the client to avoid data loss (i.e. we are changing the operation to an unregister from an unlink).

I am proposing that we do not show this warning message when `UNREG_OPR` is already set because we are already unregistering the replica at the request of the client and the warning message is just noise in that case.
